src/components/CloudHeader/CloudHeader.js

```diff
@@ -0,0 +1,164 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import classNames from 'classnames';
+import CloudHeaderWrapper from './CloudHeaderWrapper';
+import CloudHeaderMenu from './CloudHeaderMenu';
+import CloudHeaderLogo from './CloudHeaderLogo';
+import CloudHeaderList from './CloudHeaderList';
+import CloudHeaderListItem from './CloudHeaderListItem';
+
+export default class CloudHeaderV2 extends React.Component {
+  state = {
+    isMenuOpen: false,
+    isSearchActive: false,
+    isNotificationActive: false,
+    isApplicationActive: false,
+    isUserActive: false,
+  };
+
+  handleMenuClick = () => {
+    this.setState({
+      isMenuOpen: !this.state.isMenuOpen,
+    });
+  };
+
+  handleIconClick = evt => {
+    const type = evt.currentTarget.dataset.icon;
```

Comment:
Instead of reading the type info from the event, should we just partially apply the `handleIconClick` function with the given type? e.g.

```js
handleIconClick = type => event => {
 // ...
}
```

And then for the icons it'd be:

```js
<MenuIcon onClick={this.handleIconClick('menu')} />
```

Reply:
Done